src/offers/datastore/offers.js

```diff
@@ -0,0 +1,137 @@
+import Vue from 'vue'
+import offers from '@/offers/api/offers'
+import {
+  createMetaModule,
+  withMeta,
+  metaStatusesWithId,
+  createPaginationModule,
+  metaStatuses,
+} from '@/utils/datastore/helpers'
+import router from '@/base/router'
+
+const DEFAULT_STATUS = 'active'
+
+function initialState () {
+  return {
+    entries: {},
+    filter: {
+      status: DEFAULT_STATUS,
+    },
+  }
+}
+
+export default {
+  namespaced: true,
+  modules: {
+    meta: createMetaModule(),
+    pagination: createPaginationModule(),
+  },
+  state: initialState(),
+  getters: {
+    ...metaStatuses(['create']),
+    get: (state, getters) => offerId => {
+      return getters.enrich(state.entries[offerId])
+    },
+    enrich: (state, getters, rootState, rootGetters) => offer => {
+      if (!offer) return
+      return {
+        ...offer,
+        canEdit: rootGetters['auth/userId'] === offer.user,
+        user: rootGetters['users/get'](offer.user),
+        ...metaStatusesWithId(getters, ['save'], offer.id),
+      }
+    },
+    all: (state, getters) => {
+      return Object.values(state.entries)
+        .map(getters.enrich)
+    },
+    routeQuery: ({ filter: { status } }) => status === DEFAULT_STATUS ? {} : { status },
+    fetching: (state, getters) => {
+      const status = getters['meta/status']('fetchList')
+      return status.pending
+    },
+  },
+  actions: {
+    ...withMeta({
+      async fetchList ({ state, dispatch, commit }, { status = 'active' }) {
+        commit('setFilter', { status })
+        const offerList = await dispatch('pagination/extractCursor', offers.list(state.filter))
+        commit('update', offerList)
+        const users = offerList.map(offer => offer.user)
```

Comment:
Isn't this just a user id?

Reply:
Good point! I just I just blindly copied this from applications store module:

```js
        const users = applicationList.map(a => a.user)
        commit('users/update', users, { root: true })
```

I'll just remove it, I'm sure we have the users...